packages/reusable-blocks/src/components/reusable-blocks-buttons/reusable-block-delete-button.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * WordPress dependencies
+ */
+import { MenuItem } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { isReusableBlock } from '@wordpress/blocks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import { BlockSettingsMenuControls } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from '../../store/constants';
+
+export function ReusableBlockDeleteButton( { clientId } ) {
+	const { isVisible, isDisabled, block } = useSelect(
+		( select ) => {
+			const { getBlock } = select( 'core/block-editor' );
+			const { canUser } = select( 'core' );
+			const blockObj = getBlock( clientId );
```

Comment:
I hate that we have to keep inventing these awkward names to get around the outer scope lint rule 😑

Reply:
I guess splitting the selector out into a co-located hook would be a neat way around that.

Reply:
Don't mind me, talking to myself.